lib/transform.js

```diff
@@ -24,64 +25,41 @@ function addAnchor(mode, header) {
   return header;
 }
 
+function isString(y) {
+  return typeof y === 'string';
+}
 
-function getHashedHeaders (lines, maxHeaderLevel) {
-  var inCodeBlock = false
-    , lineno = 0;
 
-  // Turn all headers into '## xxx' even if they were '## xxx ##'
-  function normalize(header) {
-    return header.replace(/[ #]+$/, '');
+function getMarkdownHeaders (lines, maxHeaderLevel) {
+  function extractText (header) {
+    return header.children
+      .filter(function (x) {
+        return x.type !== 'Image';
+      })
+      .map(function (x) {
+        if (x.type === 'Link') {
+          return extractText(x);
+        }
```

Comment:
> ```javascript
> .map(function (x) {
>   if (x.type === 'Link') {
>     return extractText(x);
>   }
>   return x.raw;
> })
> ```

Wouldn't we want to recurse all the way down to the leaves? i.e.

```javascript
.map(function (x) {
  return x.type === 'Str' ? x.raw : extractText(x);
}
```

If not, what's special about links in particular?